### DISCUSSION TITLE
Joystick coordinates ode to odd

### DISCUSSION THREAD

**Discussion by MarauderRU** [logisim-evolution/logisim-evolution#1460] [Ideas]
I dont understand:
1) Why joystick coordinates start from 1 and not from 0 while all other components numbers start from 0.
2) Why coordinates range is odd. Nothing bad will happen if joystick coordinates will be shifted 1 left/up.

Digital world is even. Odd numbers confuses and make to add some crutches in scheme.

**Comment by R3dst0ne** [comment]
If you want a symmetrical joystick with a middle position, there has to be an odd number of steps.
Starting by 1 has two advantages:
- You can use the MSB to detect on what side of the middle position you are.
- The bit patterns (except MSB) are identical on both sides.


If you need the values to start at zero, simply subtract one on each output.

**Reply by MarauderRU** [reply]
Thank you for reply. I understand all of this. I use 2 bits joystick and counters to move cursor around.
But if directly connect joy to rgb screen you always lose one column and one row.
